libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManagerInternal.swift

```diff
@@ -0,0 +1,214 @@
+        
+        let codeVerifier = generateCodeVerifier()
+        guard let challenge = generateChallenge(codeVerifier: codeVerifier) else { return .unknownError }
+        let authRequestBody = "\(kSFOAuthResponseType)=\(kSFOAuthCodeCredentialsParamName)&\(kSFOAuthClientId)=\(self.clientId)&\(kSFOAuthRedirectUri)=\(redirectUri)&\(kSFOAuthCodeChallengeParamName)=\(challenge)"
+        authRequest.customHeaders = customHeaders
+        authRequest.setCustomRequestBodyString(authRequestBody, contentType: kHttpPostContentType)
+        authRequest.requiresAuthentication = false
+        authRequest.endpoint = ""
+   
+        // First REST Call - Authorization
+        let authResult = await withCheckedContinuation { continuation in
+            RestClient.sharedGlobal.send(request: authRequest) { result in
+                continuation.resume(returning: result)
+            }
+        }
+        
+        switch authResult {
+        case .success(let authResponse):
+            do {
+                let data = try authResponse.asDecodable(type: AuthorizationResponse.self)
+                let tokenRequest = RestRequest(method: RestRequest.Method.POST, baseURL: data.sfdc_community_url, path: kSFOAuthEndPointToken, queryParams: nil)
+                let tokenRequestBody = "\(kSFOAuthResponseTypeCode)=\(data.code)&\(kSFOAuthGrantType)=\(kSFOAuthGrantTypeAuthorizationCode)&\(kSFOAuthClientId)=\(self.clientId)&\(kSFOAuthRedirectUri)=\(self.redirectUri)&\(kSFOAuthCodeVerifierParamName)=\(codeVerifier)"
+                
+                tokenRequest.setCustomRequestBodyString(tokenRequestBody, contentType: kHttpPostContentType)
+                tokenRequest.requiresAuthentication = false
+                tokenRequest.endpoint = ""
+                
+                // Second REST Call - token request with code verifier
+                let tokenResult = await withCheckedContinuation { continuation in
+                    RestClient.sharedGlobal.send(request: tokenRequest) { tokenResult in
+                        continuation.resume(returning: tokenResult)
+                    }
+                }
+                
+                switch(tokenResult) {
+                case .success(let tokenResponse):
+                    UserAccountManager.shared.createNativeUserAccount(with: tokenResponse.asData(), scene:self.scene)
+                    return .success
+                case .failure(let error):
+                    SFSDKCoreLogger().e(self.classForCoder, message: "error: \(error)")
+                    return .unknownError
+                }
+            } catch {
+                SFSDKCoreLogger().e(self.classForCoder, message: "error: \(error)")
+                return .unknownError
+            }
+        case .failure(let error):
+            // You will catch the error here in the event of auth failure or if the use cannot login this way.
```

Comment:
I randomly noticed the `use` was probably supposed to be `user` in this comment.